RUFAS/biophysical/feed_storage/feed_manager.py

```diff
@@ -561,55 +561,111 @@ def _deduct_feeds_from_inventory(
-
-            if amount >= 0.001:
-                raise ValueError(f"Was not able to deduct remaining {amount} of feed {rufas_id}.")
-
-        for rufas_id, amount in total_purchased_feed_deductions.items():
-            self._om.add_variable(
-                f"purchased_feed_{rufas_id}_fed",
-                amount,
-                info_map,
-            )
-        for rufas_id, amount in total_farmgrown_feed_deductions.items():
-            self._om.add_variable(
-                f"farmgrown_feed_{rufas_id}_fed",
-                amount,
-                info_map,
-            )
+        farmgrown_by_id, purchased_by_id = self._gather_available_feeds_by_id()
+
+        total_purchased_deducted: dict[RUFAS_ID, float] = {}
+        total_farmgrown_deducted: dict[RUFAS_ID, float] = {}
+
+        for feed_id, amount_needed in feeds_to_deduct.items():
+            remaining = float(amount_needed)
+
+            remaining, farmgrown_deducted = self._deduct_from_storage(feed_id,
+                                                                      remaining,
+                                                                      farmgrown_by_id.get(feed_id, ()),
+                                                                      )
+            if farmgrown_deducted:
+                total_farmgrown_deducted[feed_id] = total_farmgrown_deducted.get(feed_id, 0.0) + farmgrown_deducted
+
+            if remaining > 1e-3:
+                remaining, purchased_deducted = \
+                    self._deduct_from_storage(feed_id, remaining, purchased_by_id.get(feed_id, ()))
+                if purchased_deducted:
+                    total_purchased_deducted[feed_id] = total_purchased_deducted.get(feed_id, 0.0) + purchased_deducted
+
+            if remaining > 1e-3:
+                self._om.add_error(
+                    "Feed Deduction Error",
+                    f"Not adequate feed to deduct remaining {remaining:.3f} kg DM of feed {feed_id}.",
+                    info_map,
+                )
+                raise ValueError(
+                    f"Not adequate feed to deduct remaining {remaining:.3f} kg DM of feed {feed_id}."
+                )
+
+        self._log_feed_deductions(total_purchased_deducted, total_farmgrown_deducted, simulation_day)
+
+    def _log_feed_deductions(
+        self,
+        total_purchased: dict[RUFAS_ID, float],
+        total_farmgrown: dict[RUFAS_ID, float],
+        simulation_day: int,
+    ) -> None:
+        """Logs the total amounts of purchased and farmgrown feeds that were deducted.
+
+        Parameters
+        ----------
+        total_purchased : dict[RUFAS_ID, float]
+            Mapping of RuFaS Feed IDs to the total amounts of that feed that were deducted from purchased feeds
+            (kg dry matter).
+        total_farmgrown : dict[RUFAS_ID, float]
+            Mapping of RuFaS Feed IDs to the total amounts of that feed that were deducted from farmgrown feeds
+            (kg dry matter).
```

Comment:
This is something that's kinda across the code base, but I think we should keep the units more consistent.
```suggestion
            Mapping of RuFaS Feed IDs to the total amounts of dry matter of that feed that were deducted from farmgrown feeds
            (kg).
```

Reply:
Yea, I thought we had a `kg DM` unit at one point - it looks like we still have a `kg CO2 / kg DM` unit but that seems problematic since the only reference to `kg CO2` is a separate unit which is `kg CO2-eq`. Maybe I will create an issue for someone to look into this. Thanks for highlighting!